app/libs/authenticate_helper.py

```diff
@@ -0,0 +1,31 @@
+import jwt
+from flask import request
+from flask_restful import abort, wraps
+
+def authentication():
+    if 'X-API-Key' not in request.headers:
+        return False
+
+    token = request.headers['X-Api-Key']
+    public_key = open('jwtRS256.key.pub').read()
+
+    try:
+        jwt.decode(token, public_key, algorithms=['RS256'])
+    except Exception:
+        return False
+
+    return True
+
+
+def authenticate(f):
```

Comment:
Authentication and Authenticate is very confusing. Would like some more clear naming for these functions.